### PR TITLE
feat(react-components): clicked node hook

### PR DESCRIPTION
**Pull request by haakonflatval-cognite**
#### Type of change

Feat  

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/UX-1147

## Description :pencil:

Adds a `useClickedNode` hook which uses `react-query` under the hood to listen to and fetch data related to Cad Node clicking.

Known issues:
- On clicking the second node after one is already clicked, the hook will spurt out three updates:
  - The first result once again
  - `undefined`
  - The result belonging to the second node

This may just be alright, but feels unnecessary

## How has this been tested? :mag:

In the story.

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [ ] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [ ] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [ ] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: react-components/stories/ClickedNode.stories.tsx
Comment (Savokr) [inline react-components/stories/ClickedNode.stories.tsx:1]: Do we need a separate story for that hook? Maybe it's fine to put it inside `HighlightNode` story in addition to what is used there and just explain the difference in comments.

**Review by Savokr — CHANGES_REQUESTED** [review]
Looks really good but a few things to consider.

File: react-components/src/hooks/useClickedNode.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    return () => {
+      viewer.off('click', callback);
+    };
+  }, [viewer]);
+
+  const nodeData = useNodeMappedData(cadIntersection?.treeIndex, cadIntersection?.model, fdmConfig);
+
+  if (nodeData === undefined || cadIntersection === undefined) {
+    return undefined;
+  }
```
[the remark this answers is not in the document]
Comment (Savokr) [reply react-components/src/hooks/useClickedNode.tsx:45]: Yeah, we can do it later. Exposing just the intersection now should also be fine I think

File: react-components/stories/ClickedNode.stories.tsx
Comment (haakonflatval-cognite) [reply react-components/stories/ClickedNode.stories.tsx:1]: Yes, okay - we will probably have too many stories eventually :sweat_smile:

Comment (haakonflatval-cognite) [reply react-components/stories/ClickedNode.stories.tsx:1]: I don't think it's necessary to include the old method of providing node data.. To be honest, this should replace it.

**Review by Savokr — APPROVED** [review]
Looks good!

**Review by Savokr — APPROVED** [review]
LGTM

**Pull request merged**